File: 2/1_Cuatrimestre/Programacion_y_Diseño_Orientado_a_Objetos/JavaFinal/NapakalakiJavaSinInterfaz/src/NapakalakiGame/CultistPlayer.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;
import java.util.ArrayList;
/**
 *
 * @author Javi
 */
public class CultistPlayer extends Player {
    private static int totalCultistPlayers=0;
    private Cultist myCultistCard;
    public CultistPlayer(Player p, Cultist c){
        super(p);
        myCultistCard=c;
    }
    public int getCombatLevel(){
        return super.getCombatLevel();
    }
    protected int getOponentLevel(Monster m){
        return m.getCombatLevelAgainstCultisPlayer();
    }
    protected boolean shouldConvert(){
        return false;
    }
    private Treasure giveMeATreasure(){
        return super.stealTreasure();
    }
    private boolean canYouGiveMeATreasure(){
        if(super.getHiddenTreasures().size()!=0){
            return true;
        }
        else{
            return false;
        }
    }
    public static int getTotalCultistPlayers(){
        return totalCultistPlayers;
    }
    public String toString(){
        return "soy un cultist   "+myCultistCard+"-->"+super.toString();
    }
}

File: 2/1_Cuatrimestre/PDOO/RubyFinal/NapakalakiRuby/lib/CultistPlayer.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative 'Player.rb'
require_relative 'Monster.rb'
module NapakalakiGame
  class CultistPlayer < Player
    @@totalCultistPlayers=0
    def CultistPlayer.totalCultistPlayers
      return @@totalCultistPlayers
    end
    attr_accessor :myCultistCard
    def initialize(p,c)
      @myCultistCard=c
      super(p.name)
      InicializarTodas(p)
    end
    def giveMeATreasure
      return super.stealTreasure
    end
    def canYouGiveMeATreasure()
      if(super.getHiddenTreasures.size!=0)then
        return true
      else
        return false
      end
    end
    def to_s
      "Soy un cultist  #{@myCultistCard}--->#{super.to_s}"
    end
    protected
    def getCombatLevel()
      return getCombatLevel
    end
    def getOponentLevel(m)
      return m.getCombatLevelAgainstCultistPlayer()
    end
    def shouldConvert
      return false
    end
  end
end
